### PR TITLE
Migrate JMX connector to new APIs

### PR DESCRIPTION
**Pull request by takezoe**

### CODE REVIEW

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxMetadata.java [hunk 1]
```diff
@@ -19,18 +19,8 @@
...
-import io.prestosql.spi.connector.ConnectorSession;
-import io.prestosql.spi.connector.ConnectorTableHandle;
-import io.prestosql.spi.connector.ConnectorTableLayout;
-import io.prestosql.spi.connector.ConnectorTableLayoutHandle;
-import io.prestosql.spi.connector.ConnectorTableLayoutResult;
-import io.prestosql.spi.connector.ConnectorTableMetadata;
-import io.prestosql.spi.connector.Constraint;
-import io.prestosql.spi.connector.SchemaTableName;
-import io.prestosql.spi.connector.SchemaTablePrefix;
+import io.prestosql.spi.connector.*;
```
Comment (findepi) [inline presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxMetadata.java]: We do not use star imports.
Hopefully importing https://github.com/airlift/codestyle into Intellij should get you the right configuration.

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxMetadata.java [hunk 2]
```diff
@@ -96,6 +87,22 @@ public JmxTableHandle getTableHandle(ConnectorSession session, SchemaTableName t
...
+    }
+
+    @Override
+    public ConnectorTableProperties getTableProperties(ConnectorSession session, ConnectorTableHandle table)
+    {
+        return new ConnectorTableProperties(TupleDomain.all(),
+            Optional.empty(),
+            Optional.empty(),
+            Optional.empty(),
+            emptyList());
```
Comment (findepi) [inline presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxMetadata.java]: 2 formatting nits (somehow this comment got lost when i first wrote it)

- per code style, when each param is on a separate line, there should be a newline before the first one:
  ```
   return new ConnectorTableProperties(
       TupleDomain.all(),
   ```
- by convention, we use `ImmutableList.of()` rather than `emptyList()` (i understand this is subjective, this is just the convention we try to adhere to, as we maintain codebase consistency)

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxMetadata.java (hunk 1 above)
Comment (takezoe) [reply presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxMetadata.java]: Thanks! Fixed it.

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxMetadata.java (hunk 2 above)
Comment (takezoe) [reply presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxMetadata.java]: Following the convention of project makes sense. I fixed them.

**Comment by takezoe** [comment]
@findepi Thanks for your review! I fixed code and updated the title of this pull request.

Currently some test cases are commented out because it seems that there is no way to give predicate to SplitManager without deprecated version of `getSplits()` method. Actually, I'm not sure how I can port this part to new APIs.

File: presto-jmx/src/test/java/io/prestosql/plugin/jmx/TestJmxSplitManager.java [hunk 3]
```diff
@@ -98,30 +98,30 @@ public void tearDown()
...
-        }
-    }
+//    @Test
+//    public void testPredicatePushdown()
+//            throws Exception
+//    {
+//        for (Node node : nodes) {
+//            String nodeIdentifier = node.getNodeIdentifier();
+//
+//            // TODO Is there any way to give predicate to SplitManager without deprecated version of getSplits()?
```
Comment (martint) [inline presto-jmx/src/test/java/io/prestosql/plugin/jmx/TestJmxSplitManager.java]: The new API doesn't yet support pushing down predicates. It's something we're adding as part of https://github.com/prestosql/presto/issues/18. We'll add a version of the API that takes a `Constraint` object like `getTableLayout` does today to ease migration. We should probably wait to complete and merge this PR until that gets in (which will happen soon), otherwise, we're removing existing features for this connector.

Comment (takezoe) [reply presto-jmx/src/test/java/io/prestosql/plugin/jmx/TestJmxSplitManager.java]: Thanks for the explanation! I understand it wasn't a time to migrate connectors yet. I'm waiting for `Constraint` to be added.

Comment (martint) [reply presto-jmx/src/test/java/io/prestosql/plugin/jmx/TestJmxSplitManager.java]: @takezoe, with this, you should be able to complete the migration of this connector: https://github.com/prestosql/presto/pull/541.

**Comment by martint** [comment]
@takezoe, https://github.com/prestosql/presto/pull/541 is merged, so you should be able to rebase this PR and implement the missing functionality now.

**Comment by takezoe** [comment]
Thanks! 'm going to resume working on this pull request.

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxTableHandle.java [hunk 4]
```diff
@@ -114,4 +126,60 @@ public ConnectorTableMetadata getTableMetadata()
                 tableName,
                 ImmutableList.copyOf(transform(columnHandles, JmxColumnHandle::getColumnMetadata)));
     }
+
+    public static class Node
+    {
+        private String identifier;
+        private HostAddress address;
```
Comment (ebyhr) [inline presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxTableHandle.java]: I think these fields should be `private final`.

File: presto-jmx/src/test/java/io/prestosql/plugin/jmx/TestJmxMetadata.java [hunk 5]
```diff
@@ -96,4 +114,32 @@ public void testGetCumulativeTableHandle()
...
     }
+
+    @Test
+    public void testApplyFilterWithoutFilter()
+    {
+        JmxTableHandle handle = metadata.getTableHandle(SESSION, new SchemaTableName(JMX_SCHEMA_NAME, "java.lang:*"));
+        Optional<ConstraintApplicationResult<ConnectorTableHandle>> result = metadata.applyFilter(handle, new Constraint(TupleDomain.all()));
+
+        assertTrue(result.isPresent());
+        assertTrue(((JmxTableHandle) result.get().getHandle()).getNodes().size() == 3);
```
Comment (ebyhr) [inline presto-jmx/src/test/java/io/prestosql/plugin/jmx/TestJmxMetadata.java]: It would be better to use `assertEquals`. L138 is also.

File: presto-jmx/src/test/java/io/prestosql/plugin/jmx/TestJmxTableHandle.java [hunk 6]
```diff
@@ -48,23 +50,31 @@ public void testEquivalence()
...
+                        new JmxTableHandle(SCHEMA_TABLE_NAME, ImmutableList.of("name"), singleColumn, false, ImmutableList.of()))
+                .addEquivalentGroup(
+                        new JmxTableHandle(SCHEMA_TABLE_NAME, ImmutableList.of("name"), singleColumn, false, ImmutableList.of(createTestingNode("host1"))),
+                        new JmxTableHandle(SCHEMA_TABLE_NAME, ImmutableList.of("name"), singleColumn, false, ImmutableList.of(createTestingNode("host1"))))
                 .check();
     }
+
+    private static JmxTableHandle.Node createTestingNode(String hostname)
+    {
+        return new JmxTableHandle.Node(hostname, HostAddress.fromString(format("%s:8080", hostname)));
```
Comment (ebyhr) [inline presto-jmx/src/test/java/io/prestosql/plugin/jmx/TestJmxTableHandle.java]: `HostAddress.fromParts` is simpler than concating hostname and port.
```java
return new JmxTableHandle.Node(hostname, HostAddress.fromParts(hostname, 8080));
```

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxSplitManager.java [hunk 7]
```diff
@@ -14,64 +14,29 @@
...
 
-        List<ConnectorSplit> splits = nodeManager.getAllNodes().stream()
-                .filter(node -> {
-                    NullableValue value = NullableValue.of(createUnboundedVarcharType(), utf8Slice(node.getNodeIdentifier()));
-                    return predicate.overlaps(fromFixedValues(ImmutableMap.of(nodeColumnHandle.get(), value)));
-                })
-                .map(node -> new JmxSplit(ImmutableList.of(node.getHostAndPort())))
-                .collect(toList());
+        List<ConnectorSplit> splits = tableHandle.getNodes().stream()
+                .map(node -> new JmxSplit(ImmutableList.of(node.getAddress()))).collect(toList());
```
Comment (ebyhr) [inline presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxSplitManager.java]: Move `.collect(toList());` to next line.

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxRecordSetProvider.java [hunk 8]
```diff
@@ -153,7 +153,12 @@ else if (value.getClass().getComponentType() == short.class) {
     }
 
     @Override
-    public RecordSet getRecordSet(ConnectorTransactionHandle transaction, ConnectorSession session, ConnectorSplit split, ConnectorTableHandle table, List<? extends ColumnHandle> columns)
+    public RecordSet getRecordSet(
+            ConnectorTransactionHandle transaction,
+            ConnectorSession session,
+            ConnectorSplit split,
+            ConnectorTableHandle table,
+            List<? extends ColumnHandle> columns)
```
Comment (ebyhr) [inline presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxRecordSetProvider.java]: Unnecessary change for jmx migration. Basically, we separate the commit to fix styles.

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxTableHandle.java [hunk 9]
```diff
@@ -105,6 +116,7 @@ public String toString()
                 .add("objectNames", objectNames)
```
Comment (ebyhr) [inline presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxTableHandle.java]: Found existing code has wrong combination in `toString()`.
```
.add("tableName", objectNames)
```

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxMetadata.java [hunk 10]
```diff
@@ -242,17 +271,28 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
...
+        TupleDomain<ColumnHandle> predicate = constraint.getSummary();
+
+        List<JmxTableHandle.Node> nodes = tableHandle.getNodes().stream()
+                .filter(node -> {
+                    NullableValue value = NullableValue.of(createUnboundedVarcharType(), utf8Slice(node.getIdentifier()));
+                    return predicate.overlaps(fromFixedValues(ImmutableMap.of(nodeColumnHandle.get(), value)));
+                })
+                .collect(toList());
+
+        JmxTableHandle newTaleHandle = new JmxTableHandle(tableHandle.getTableName(), tableHandle.getObjectNames(), tableHandle.getColumnHandles(), tableHandle.isLiveData(), nodes);
```
Comment (ebyhr) [inline presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxMetadata.java]: Typo `newTaleHandle`

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxRecordSetProvider.java (hunk 8 above)
Comment (takezoe) [reply presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxRecordSetProvider.java]: I can't remember. Let me revert this change.

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxSplitManager.java (hunk 7 above)
Comment (takezoe) [reply presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxSplitManager.java]: > Move .collect(toList()); to next line.

Do we have a specific regulation about putting new lines? Actually, I have a similar code in `JmxMetadata` as follows. I'm wondering if I have to put new lines before `.map` and `.collect`.

```scala
private List<JmxTableHandle.Node> getAllNodes()
{
    return nodeManager.getAllNodes().stream().map(node -> new JmxTableHandle.Node(node.getNodeIdentifier(), node.getHostAndPort())).collect(toList());
}
```

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxTableHandle.java (hunk 9 above)
Comment (takezoe) [reply presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxTableHandle.java]: Nice catch! I will create another pull request to fix it.

Comment (takezoe) [reply presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxTableHandle.java]: Created: https://github.com/prestosql/presto/pull/634

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxSplitManager.java (hunk 7 above)
Comment (ebyhr) [reply presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxSplitManager.java]: When the statement is simple, one line is fine. When you break a line after `.stream()`, subsequent each operations should be independent line as original code was. (There's no 'specific' regulation as far as I know).

The `getAllNodes` looks little long for one line and there's only one statement in the method, so putting new lines looks reasonable to me.

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxTableHandle.java (hunk 9 above)
Comment (ebyhr) [reply presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxTableHandle.java]: Thank you for your quick fix!

**Comment by takezoe** [comment]
@martint I finished to migrate the JMX connector node selection pushdown to `Metadata.applyFilter`. Can you take a look?

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxTableHandle.java [hunk 11]
```diff
@@ -36,21 +36,21 @@
...
+    private final List<Node> nodes;
 
     @JsonCreator
     public JmxTableHandle(
             @JsonProperty("tableName") SchemaTableName tableName,
             @JsonProperty("objectNames") List<String> objectNames,
             @JsonProperty("columnHandles") List<JmxColumnHandle> columnHandles,
             @JsonProperty("liveData") boolean liveData,
-            @JsonProperty("addresses") List<HostAddress> addresses)
+            @JsonProperty("nodes") List<Node> nodes)
```
Comment (martint) [inline presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxTableHandle.java]: Why hold on to the list of nodes in the TableHandle? I'd be better to just hold on to the predicate and filter them out during split generation.

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxMetadata.java [hunk 12]
```diff
@@ -96,6 +103,23 @@ public JmxTableHandle getTableHandle(ConnectorSession session, SchemaTableName t
...
+    @Override
+    public boolean usesLegacyTableLayouts()
+    {
+        return false;
+    }
+
+    @Override
+    public ConnectorTableProperties getTableProperties(ConnectorSession session, ConnectorTableHandle table)
+    {
+        return new ConnectorTableProperties(
```
Comment (electrum) [inline presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxMetadata.java]: This can use the no-args constructor, which does the same thing

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxTableHandle.java (hunk 11 above)
Comment (takezoe) [reply presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxTableHandle.java]: In my understanding, to hold the predicate ( `TupleDomain` or `Map<ColumnHandle, Domain>`) in the TableHandle, we need to make all classes held in the `ValueSet` serializable to JSON. Is this correct?

Comment (takezoe) [reply presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxTableHandle.java]: Oops, I didn't know `TupleDomain` is always JSON serializable. I fixed to hold `TupleDomain` in `TableHandle` instead of the list of node.

**Comment by martint** [comment]
@takezoe, can you rebase on master and remove the merge commits? They are confusing github and it's not displaying the actual differences with respect to the base commit.

**Comment by takezoe** [comment]
Sure, I rebased on master branch without merge commits.

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxMetadata.java [hunk 13]
```diff
@@ -242,17 +254,14 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
...
     @Override
-    public List<ConnectorTableLayoutResult> getTableLayouts(ConnectorSession session, ConnectorTableHandle table, Constraint constraint, Optional<Set<ColumnHandle>> desiredColumns)
+    public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(ConnectorSession session, ConnectorTableHandle handle, Constraint constraint)
     {
-        JmxTableHandle handle = (JmxTableHandle) table;
-        ConnectorTableLayout layout = new ConnectorTableLayout(new JmxTableLayoutHandle(handle, constraint.getSummary()));
-        return ImmutableList.of(new ConnectorTableLayoutResult(layout, constraint.getSummary()));
-    }
+        JmxTableHandle tableHandle = (JmxTableHandle) handle;
+        TupleDomain<ColumnHandle> predicate = constraint.getSummary();
```
Comment (martint) [inline presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxMetadata.java]: `applyFilter` can get called multiple times throughout the optimization process. Connectors need to ensure that they "remember" the result of previous applications. Also, if applying the filter did not have any effect, the method needs to return `Optional.empty()`. Otherwise, the optimizer may loop forever.

See https://github.com/prestosql/presto/blob/master/presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcMetadata.java#L86 for an example.

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxMetadata.java [hunk 14]
```diff
@@ -242,17 +254,14 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
...
 
-    @Override
-    public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTableLayoutHandle handle)
-    {
-        return new ConnectorTableLayout(handle);
+        JmxTableHandle newTableHandle = new JmxTableHandle(tableHandle.getTableName(), tableHandle.getObjectNames(), tableHandle.getColumnHandles(), tableHandle.isLiveData(), predicate);
+
+        return Optional.of(new ConstraintApplicationResult(newTableHandle, predicate));
```
Comment (martint) [inline presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxMetadata.java]: If the connector guarantees the results are going to satisfy the constraint, it's not necessary to return the predicate in the `ConstraintApplicationResult`. It can just indicate `TupleDomain.ALL`.

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxTableHandle.java [hunk 15]
```diff
@@ -35,18 +37,21 @@
...
             @JsonProperty("columnHandles") List<JmxColumnHandle> columnHandles,
-            @JsonProperty("liveData") boolean liveData)
+            @JsonProperty("liveData") boolean liveData,
+            @JsonProperty("constraint") TupleDomain<ColumnHandle> nodeFilter)
     {
         this.tableName = requireNonNull(tableName, "tableName is null");
         this.objectNames = ImmutableList.copyOf(requireNonNull(objectNames, "objectName is null"));
         this.columnHandles = ImmutableList.copyOf(requireNonNull(columnHandles, "columnHandles is null"));
         this.liveData = liveData;
+        this.nodeFilter = requireNonNull(nodeFilter, "constraint is null");
```
Comment (ebyhr) [inline presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxTableHandle.java]: I guess `nodeFilter is null` is expected message and L48 should be updated `@JsonProperty("nodeFilter")`.

Comment (takezoe) [reply presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxTableHandle.java]: Oh, that's true!

**Comment by takezoe** [comment]
@martint Thanks for your review again. Fixed to return `Option.empty` if `JmxTableHandle` has a exact same constraint. Also now `JmxTableHandle` holds only node column filter and others are passed to `ConstraintApplicationResult` as `remainingFilter` to exclude predicates which are processed by the connector.

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxSplitManager.java [hunk 16]
```diff
@@ -52,22 +52,22 @@ public JmxSplitManager(NodeManager nodeManager)
...
     @Override
-    public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, ConnectorSession session, ConnectorTableLayoutHandle layout, SplitSchedulingStrategy splitSchedulingStrategy)
+    public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, ConnectorSession session, ConnectorTableHandle table, SplitSchedulingStrategy splitSchedulingStrategy)
     {
-        JmxTableLayoutHandle jmxLayout = (JmxTableLayoutHandle) layout;
-        JmxTableHandle tableHandle = jmxLayout.getTable();
-        TupleDomain<ColumnHandle> predicate = jmxLayout.getConstraint();
+        JmxTableHandle tableHandle = (JmxTableHandle) table;
 
-        //TODO is there a better way to get the node column?
```
Comment (martint) [inline presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxSplitManager.java]: unrelated change?

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxSplitManager.java [hunk 17]
```diff
@@ -52,22 +52,22 @@ public JmxSplitManager(NodeManager nodeManager)
...
         Optional<JmxColumnHandle> nodeColumnHandle = tableHandle.getColumnHandles().stream()
                 .filter(jmxColumnHandle -> jmxColumnHandle.getColumnName().equals(NODE_COLUMN_NAME))
                 .findFirst();
+
```
Comment (martint) [inline presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxSplitManager.java]: Unrelated formatting change.

File: presto-jmx/src/test/java/io/prestosql/plugin/jmx/TestJmxTableHandle.java [hunk 18]
```diff
@@ -48,23 +56,26 @@ public void testEquivalence()
...
-                        new JmxTableHandle(SCHEMA_TABLE_NAME, ImmutableList.of("name"), COLUMNS, true))
+                        new JmxTableHandle(SCHEMA_TABLE_NAME, ImmutableList.of("name"), COLUMNS, true, TupleDomain.all()),
+                        new JmxTableHandle(SCHEMA_TABLE_NAME, ImmutableList.of("name"), COLUMNS, true, TupleDomain.all()))
+                .addEquivalentGroup(
+                        new JmxTableHandle(SCHEMA_TABLE_NAME, ImmutableList.of("name"), COLUMNS, false, TupleDomain.all()),
+                        new JmxTableHandle(SCHEMA_TABLE_NAME, ImmutableList.of("name"), COLUMNS, false, TupleDomain.all()))
                 .addEquivalentGroup(
-                        new JmxTableHandle(SCHEMA_TABLE_NAME, ImmutableList.of("name"), COLUMNS, false),
-                        new JmxTableHandle(SCHEMA_TABLE_NAME, ImmutableList.of("name"), COLUMNS, false))
+                        new JmxTableHandle(SCHEMA_TABLE_NAME, ImmutableList.of("nameX"), COLUMNS, true, TupleDomain.all()),
```
Comment (martint) [inline presto-jmx/src/test/java/io/prestosql/plugin/jmx/TestJmxTableHandle.java]: Why the name change here?

File: presto-jmx/src/test/java/io/prestosql/plugin/jmx/TestJmxTableHandle.java [hunk 19]
```diff
@@ -48,23 +56,26 @@ public void testEquivalence()
...
+                        new JmxTableHandle(SCHEMA_TABLE_NAME, ImmutableList.of("nameX"), COLUMNS, true, TupleDomain.all()))
                 .addEquivalentGroup(
-                        new JmxTableHandle(SCHEMA_TABLE_NAME, ImmutableList.of("nameX"), COLUMNS, true),
-                        new JmxTableHandle(SCHEMA_TABLE_NAME, ImmutableList.of("nameX"), COLUMNS, true))
+                        new JmxTableHandle(SCHEMA_TABLE_NAME, ImmutableList.of("nameX"), COLUMNS, false, TupleDomain.all()),
+                        new JmxTableHandle(SCHEMA_TABLE_NAME, ImmutableList.of("nameX"), COLUMNS, false, TupleDomain.all()))
                 .addEquivalentGroup(
-                        new JmxTableHandle(SCHEMA_TABLE_NAME, ImmutableList.of("nameX"), COLUMNS, false),
```
Comment (martint) [inline presto-jmx/src/test/java/io/prestosql/plugin/jmx/TestJmxTableHandle.java]: Why the name change here?

File: presto-jmx/src/test/java/io/prestosql/plugin/jmx/TestJmxTableHandle.java (hunk 18 above)
Comment (takezoe) [reply presto-jmx/src/test/java/io/prestosql/plugin/jmx/TestJmxTableHandle.java]: This is diff magic. I didn't change the name, just added new arguments. "name" patterns are coverd by preceding 2 test cases.

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxSplitManager.java (hunk 16 above)
Comment (takezoe) [reply presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxSplitManager.java]: Oh, sorry. Reverted.

File: presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxSplitManager.java (hunk 17 above)
Comment (takezoe) [reply presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxSplitManager.java]: True. Reverted too.

**Comment by takezoe** [comment]
@martint Thanks for your comment! Fixed and squashed.

**Comment by martint** [comment]
Merged. Thanks for your contribution!
